File: CSharp/MVC/SampleSite/Controllers/OpenCadesSignatureController.cs
using Lacuna.RestPki.Api;
using Lacuna.RestPki.Client;
using SampleSite.Classes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.Web;
using System.Web.Mvc;

namespace Lacuna.RestPki.SampleSite.Controllers {

	public class OpenCadesSignatureController : Controller {

		/**
		 * This action submits a CAdES signature file to REST PKI for inspection.
		 */
		[HttpGet]
		public async Task<ActionResult> Index(string userfile) {

			// Our action only works if a userfile is given to work with.
			if (string.IsNullOrEmpty(userfile)) {
				return HttpNotFound();
			}
            var filename = userfile.Replace("_", ".");
            // Note: we're receiving the userfile argument with "_" as "." because of limitations of
            // ASP.NET MVC.

            // Get an instance of the CadesSignatureExplorer class, used to open/validate CAdES
            // signatures.
            var sigExplorer = new CadesSignatureExplorer(Util.GetRestPkiClient()) {
                // Specify that we want to validate the signatures in the file, not only inspect them.
                Validate = true,
                // Specify the parameters for the signature validation:
                // Accept any CAdES signature as long as the signer has an ICP-Brasil certificate.
                DefaultSignaturePolicyId = StandardCadesSignaturePolicies.CadesBes,
                // We have encapsulated the security context choice on Util.cs
                SecurityContextId = Util.GetSecurityContextId()
            };

			// Set the CAdES signature file
			sigExplorer.SetSignatureFile(Server.MapPath("~/App_Data/" + filename));

			// Call the Open() method, which returns the signature file's information
			var signature = await sigExplorer.OpenAsync();

            // Render the information. (see file Views/OpenCadesSignature/Index.html for more information on
            // the information returned)
			return View(signature);
		}
	}
}

File: Java/sample-spring-mvc/src/main/java/sample/controller/OpenCadesSignatureController.java
package sample.controller;

import com.lacunasoftware.restpki.*;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import sample.Application;
import sample.util.Util;

import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

@Controller
public class OpenCadesSignatureController {

	/**
	 * This action submits a CAdES signature file to Rest PKI for inspection.
	 */
	@RequestMapping(value = "/open-cades-signature", method = {RequestMethod.GET})
	public String get(
			@RequestParam(value = "userfile", required = false) String userfile,
			Model model
	) throws IOException, RestException {

		// Get an instance of the CadesSignatureExplorer2 class, used to open/validate CAdES
		// signatures.
		CadesSignatureExplorer2 sigExplorer = new CadesSignatureExplorer2(Util.getRestPkiClient());

		// Specify that we want to validate the signatures in the file, not only inspect them.
		sigExplorer.setValidate(true);

		// Specify the parameters for the signature validation:
		// Accept any CAdES signature as long as the signer has an ICP-Brasil
		sigExplorer.setDefaultSignaturePolicy(SignaturePolicy.CadesBes);
		// We have encapsulated the security context choice on Util.java.
		sigExplorer.setSecurityContext(Util.getSecurityContextId());

		// Alternatively, you may require full compliance with ICP-Brasil by doing:
		//sigExplorer.setAcceptableExplicitPolicies(SignaturePolicyCatalog.getPkiBrazilCades());
		//sigExplorer.setDefaultSignaturePolicy(null);
		//sigExplorer.setSecurityContext(null);

		// Set the CAdES signature file.
		sigExplorer.setSignatureFile(Application.getTempFolderPath().resolve(userfile));

		// Call the open() method, which returns the signature file's information.
		CadesSignature signature = sigExplorer.open();

		// Render the information (see file resources/templates/open-cades-signature.html for more
		// information on the information returned).
		model.addAttribute("signature", signature);

		return "open-cades-signature";
	}
}
